### DISCUSSION TITLE
error to retrive data

### DISCUSSION THREAD

**Discussion by sabhattrai** [Cyfrin/foundry-full-course-cu#1696] [General]
how to use cast call to retrive a string data from a function.
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/121299789/5a01769f-9e82-4123-8a43-e191fda1623f)
got an output 
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/121299789/13ccff7f-24b8-4098-8549-124bc52071a9)

**Comment by EngrPips** [comment]
Hello @sabhattrai, This contract you are trying to read value from is deployed to which blockchain?

**Reply by sabhattrai** [reply]
sepolia testnet

**Reply by sabhattrai** [reply]
and also how to verify the contract while deploying the contract . while using --verify i got an error " Missing etherscan key for chain 11155111"
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/121299789/69bcc3e7-ca16-4a19-b4c3-31c9eb2ffc3e)
got an error
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/121299789/220627c8-81af-4b2c-ad34-1d821e7ae68c)

**Reply by EngrPips** [reply]
To interact with an already deployed contract you should use the command below 

```js 
cast send <the address of the contract you want to interact with> <function signature of the method to call at the contract address> <argument that should pass along> --rpc-url <http://your_rpc_url> --private-key <your private key>
```

To verify a contract on etherscan at the same time you deploy it you need to actually have a correct `etherscan api key` in your `.env`
